lib/plugins/bower-manifest.js

```diff
@@ -0,0 +1,61 @@
+import processJSON from './helper/process-json';
+import { isSemver } from './helper/version';
+import JavaScriptPlugin from './javascript';
+import { jsonRegExKeyValue, jsonRegExValue } from './helper/regex-builder';
+import insertLink from '../insert-link';
+import liveResolverQuery from '../resolver/live-resolver-query.js';
+import gitUrl from '../resolver/git-url.js';
+import githubShorthand from '../resolver/github-shorthand.js';
+
+function linkDependency(type, blob, key, value) {
+  const isValidSemver = isSemver(value);
+  const regex = jsonRegExKeyValue(key, value);
+
+  insertLink(blob.el, regex, {
+    pluginName: 'BowerManifest',
+    target: isValidSemver ? '$1' : '$2',
+    type: isValidSemver ? 'liveResolverQuery' : 'git',
+  });
+}
+
+function linkFile(blob, key, value) {
+  const regex = jsonRegExValue(key, value);
+
+  insertLink(blob.el, regex, {
+    pluginName: 'BowerManifest',
+    type: 'file',
+    path: blob.path,
+    target: '$1',
+  });
+}
+
+export default class BowerManifest {
+
+  static resolve({ target, path, type }) {
+    if (type === 'file') {
+      return JavaScriptPlugin.resolve({ target, path });
+    }
+
+    if (type === 'liveResolverQuery') {
+      return liveResolverQuery({ type: 'bower', target });
+    }
+
+    return [
+      gitUrl({ target }),
+      githubShorthand({ target }),
+    ];
+  }
```

Comment:
Are you talking about the calls to `JavaScriptPlugin.resolve`, `liveResolverQuery`, `gitUrl`, and `githubShorthand` being made explicitly, rather than implicitly like they are [here](https://github.com/OctoLinker/browser-extension/pull/274/files/e24c1ddec2445790d439dddfa09384ece592a520#diff-a8fe7166d47dcb744448dab0f2cf7befL11) and [here](https://github.com/OctoLinker/browser-extension/pull/274/files/e24c1ddec2445790d439dddfa09384ece592a520#diff-a8fe7166d47dcb744448dab0f2cf7befL21)? That doesn't really seem like a problem to me, but perhaps I'm misunderstanding the question. Splitting the Bower plugin into multiple other plugins sounds like it would just make things more complicated, especially if it's done as part of this refactor, rather than as a separate change.